code/src/java/pcgen/rules/context/AbstractReferenceContext.java

```diff
@@ -127,13 +128,17 @@ public boolean validate(UnconstructedValidator validator)
 		return getManufacturer(c, cat).getTypeReference(val);
 	}
 
-	public <T extends Loadable> T constructCDOMObject(Class<T> c, String val)
+	public <T extends Loadable, U extends Loadable & CategorizedCDOMObject<U>> T constructCDOMObject(Class<T> c, String val)
```

Comment:
The complexity provides error catching, in the sense that instead of generics warnings, we get the compiler checking that we have the correct object type.